### ISSUE TITLE
Legion: assertion error at termination

### ISSUE THREAD

**Issue by bandokihiro** [StanfordLegion/legion#934]
I am hitting this assertion error when my code finishes:
```
solver: /home/bandokihiro/Softwares/legion_mod/runtime/legion/region_tree.cc:14360: void Legion::Internal::FieldSpaceNode::process_allocator_flush(Legion::Deserializer&): Assertion `allocation_state == FIELD_ALLOC_PENDING' failed.
Legion process received signal 6: Aborted
```
I reproduced it with the most recent `control_replication` commit f317a05dbdc14829abdb47ea13d7d8c001a49da7. Both legion and my code are compiled in debug mode (this doesn't happen in release mode). The command line I used on sapling is
```
mpirun -n 3 -H n0000,n0001,n0002 -npernode 1 -bind-to none -x LEGION_FREEZE_ON_ERROR /home/bandokihiro/DG-Legion/build_master_debug/exec/solver -logfile logs/log_%.log -hdf5:forcerw -ll:cpu 1 -ll:util 2 -ll:dma 2 -ll:onuma 0 -ll:ocpu 1 -ll:othr 5 -ll:csize 1000 -ll:rsize 0
```

**Comment by bandokihiro** [comment]
[gdb.txt](https://github.com/StanfordLegion/legion/files/5256689/gdb.txt)
Here is a backtrace.

**Comment by lightsighter** [comment]
Doesn't reproduce. I get an error about not finding some kind of file:

```
mebauer@sapling:~$ mpirun -n 3 -H n0000,n0001,n0002 -npernode 1 -bind-to none -x LEGION_FREEZE_ON_ERROR /home/bandokihiro/DG-Legion/build_master_debug/exec/solver -logfile logs/log_%.log -hdf5:forcerw -ll:cpu 1 -ll:util 2 -ll:dma 2 -ll:onuma 0 -ll:ocpu 1 -ll:othr 5 -ll:csize 1000 -ll:rsize 0
[sapling:25229] Warning: could not find environment variable "LEGION_FREEZE_ON_ERROR"
terminate called after throwing an instance of 'std::runtime_error'
  what():  toml::parse: file open error -> input.toml
terminate called after throwing an instance of 'std::runtime_error'
  what():  toml::parse: file open error -> input.toml
--------------------------------------------------------------------------
Primary job  terminated normally, but 1 process returned
a non-zero exit code. Per user-direction, the job has been aborted.
--------------------------------------------------------------------------
terminate called after throwing an instance of 'std::runtime_error'
  what():  toml::parse: file open error -> input.toml
--------------------------------------------------------------------------
mpirun noticed that process rank 1 with PID 8887 on node n0001 exited on signal 6 (Aborted).
--------------------------------------------------------------------------
2 total processes killed (some possibly by mpirun during cleanup)
```

**Comment by bandokihiro** [comment]
Please go to `/home/bandokihiro/Public/DebugTermination/run` and run `run.sh`. I noticed that this issue occurs when using 3 sub-regions/ranks and not with 2 or 4.

**Comment by bandokihiro** [comment]
I am also having other issues with 3 sub-regions/ranks so something seems to be going wrong with that particular number

**Comment by bandokihiro** [comment]
This bug looks fixed on f96e7d2562db0abbeda05f60237e2bf1456004c5 so I am going to close this.
